sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,302 @@
+    streaming_interface_remove_downstream_stall;
+
+inline constexpr register_map_interface_key::value_t<
+  register_map_interface_options_enum::wait_for_done_write>
+    register_map_interface_wait_for_done_write;
+
+inline constexpr register_map_interface_key::value_t<
+  register_map_interface_options_enum::do_not_wait_for_done_write>
+    register_map_interface_do_not_wait_for_done_write;
+
+inline constexpr register_map_interface_key::value_t<option>
+  register_map_interface;
+
+template<int initiation_interval_value>
+inline constexpr initiation_interval_key::value_t<initiation_interval_value>
+  initiation_interval;
+
+template<bool pipeline_directive, typename initiation_interval>
+inline constexpr pipelined_key::value_t<pipeline_directive,
+  initiation_interval> pipelined;
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+|===
+|Property|Description
+|`streaming_interface`
+|The `streaming_interface` property adds the requirement that the kernel must
+ have dedicated ports for input / output signals. This applies for both
+ control, and kernel argument data signals. The following values are supported:
+
+ * `accept_downstream_stall`: Directs the compiler to generate a kernel
+    interface that can accept a back-pressure signal.
+
+ * `remove_downstream_stall`: Directs the compiler to generate a kernel
+    interface that does not accept a back-pressure signal.
+
+ If the `streaming_interface` property is not specified, the default behavior is
+ equivalent to one of the values listed above, but the choice is implementation
+ defined.
+
+ The following properties have been provided for convenience:
+ `streaming_interface_accept_downstream_stall`,
+ `streaming_interface_remove_downstream_stall`.
+
+|`register_map_interface`
+|The `register_map_interface` property adds the requirement that the kernel must
+ have its input / output control and kernel argument data signals placed in a
+ shared Control and Status Register (CSR) map. The following values are
+ supported:
+
+ * `wait_for_done_write`: Directs the compiler to generate logic that
+    back-pressures the kernel until the kernel is notified that its completion
+    has been detected. The kernel will be notified when the register it writes
+    its completion signal to, is set to 1.
+
+ * `do_not_wait_for_done_write`: Directs the compiler to not generate logic that
+    would back-pressure the kernel until the kernel is notified of its
+    completion being detected.
+ 
+ If the `register_map_interface` property is not specified, the default behavior
+ is equivalent to one of the values listed above, but the choice is
+ implementation defined.
+
+ The following properties have been provided for convenience:
+ `register_map_interface_wait_for_done_write`,
+ `register_map_interface_do_not_wait_for_done_write`.
+
+|`pipelined`
+|Two sets of values are accepted as property parameters - a boolean, and another
+ property `initiation_interval`.
```

Comment:
Is there a benefit to defining `initiation_interval` as a property? I worry that having it use the property interface may wrongfully give the user the idea that it is to be used like a property, while it is simply intended for use inside the `pipelined` property. It also adds more to the implementation, as properties need to define a number of things under the hood, whereas if we define it as some simple wrapper class we can avoid that.

Reply:
Hi Steffen, I have updated the spec, please review again. Thank you!